src/XMakeBuildEngine/BackEnd/BuildManager/BuildParameters.cs

```diff
@@ -1007,56 +1003,16 @@ private void FindMSBuildExe()
             EnsureToolsets();
             Toolset currentToolset = _toolsetProvider.GetToolset(MSBuildConstants.CurrentToolsVersion);
 
-            if (currentToolset != null)
+            if (currentToolset != null && !string.IsNullOrEmpty(currentToolset.ToolsPath))
             {
-                path = Path.Combine(currentToolset.ToolsPath, "MSBuild.exe");
-                if (path != null && CheckMSBuildExeExistsAt(path))
+                location = Path.Combine(currentToolset.ToolsPath, "MSBuild.exe");
+                if (CheckMSBuildExeExistsAt(location))
                 {
-                    _nodeExeLocation = path;
-                    return;
+                    return location;
                 }
             }
 
-            // Search in the location of any assemblies we have loaded.
-            foreach (System.Reflection.Assembly assembly in AppDomain.CurrentDomain.GetAssemblies())
```

Comment:
The problem here is the first assembly is usually mscorlib.dll. MSBuild.exe exists in that location, so it would always use that. We could filter out those, but I don't think it's needed since it wouldn't have worked since MSBuild was actually shipped in the Framework 8 years ago or whatever :smile: